website/project/decorators.py

```diff
@@ -41,19 +41,45 @@ def _kwargs_to_nodes(kwargs):
     return project, node
 
 
-def must_be_valid_project(func):
+def must_be_valid_project(func=None, are_retractions_valid=False):
+    """ Ensures permissions to retractions are never implicitly granted. """
 
     # TODO: Check private link
+    def must_be_valid_project_inner(func):
+
+        @functools.wraps(func)
+        def wrapped(*args, **kwargs):
+
+            kwargs['project'], kwargs['node'] = _kwargs_to_nodes(kwargs)
+            if not are_retractions_valid and getattr(kwargs['project'].retraction, 'is_retracted', False):
+                raise HTTPError(http.BAD_REQUEST)
+            elif kwargs['node'] and not are_retractions_valid and getattr(kwargs['node'].retraction, 'is_retracted', False):
+                raise HTTPError(http.BAD_REQUEST)
+            else:
+                return func(*args, **kwargs)
+
+        return wrapped
+
+    if func:
+        return must_be_valid_project_inner(func)
+
+    return must_be_valid_project_inner
+
+
+def must_be_public_registration(func):
 
     @functools.wraps(func)
     def wrapped(*args, **kwargs):
 
-        kwargs['project'], kwargs['node'] = _kwargs_to_nodes(kwargs)
```

Comment:
This is just git being wonky. It wasn't changed, that's part of a new decorator. The old one was adjusted and tests were written for it